dask/dataframe/io/parquet/core.py

```diff
@@ -95,6 +97,74 @@ def __call__(self, part):
         )
 
 
+class ToParquetFunctionWrapper:
+    """
+    Parquet Function-Wrapper Class
+
+    Writes a DataFrame partition into a distinct parquet
+    file. When called, the function also requires the
+    current block index (via ``blockwise.BlockIndex``).
+    """
+
+    def __init__(
+        self,
+        engine,
+        path,
+        fs,
+        partition_on,
+        write_metadata_file,
+        i_offset,
+        name_function,
+        kwargs_pass,
+    ):
+        self.engine = engine
+        self.path = path
+        self.fs = fs
+        self.partition_on = partition_on
+        self.write_metadata_file = write_metadata_file
+        self.i_offset = i_offset
+        self.name_function = name_function
+        self.kwargs_pass = kwargs_pass
+
+    def __repr__(self):
+        # This repr must begin with "to-parquet"
+        # for the name of the resulting `Blockwise`
+        # layer to begin with "to-parquet-"
+        return "to-parquet"
+
+    def __dask_tokenize__(self):
+        return tokenize(
```

Comment:
```suggestion
        return (
```
Minor note: I _think_ it's not actually necessary to tokenize all these things ourselves; you can just return the normalized form whose string repr you want to be md5'd
https://github.com/dask/dask/blob/a504115312b1e591604276a3098b2e0076bf49e1/dask/base.py#L855

Doesn't matter at all though